### PR TITLE
fix search functionality.Add sort functionality

### PR DESCRIPTION
**Pull request by valeskus**

### CODE REVIEW

File: src/stores/search/hooks/useSetSearch.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import * as React from 'react';
+import * as Redux from 'react-redux';
+
+import { setSearch } from '../searchActions';
+
+import { SearchOptions } from 'src/models';
+
+export const useSetSearch = () => {
```
Comment (MAKARD) [inline src/stores/search/hooks/useSetSearch.ts]: useSetSearchOptions

File: src/stores/search/hooks/useSetSearch.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import * as React from 'react';
+import * as Redux from 'react-redux';
+
+import { setSearch } from '../searchActions';
+
+import { SearchOptions } from 'src/models';
```
Comment (MAKARD) [inline src/stores/search/hooks/useSetSearch.ts]: use relative import

File: src/UI/PickListItem/PickListItem.tsx [hunk 3]
```diff
@@ -7,32 +7,26 @@ import { styles } from './styles';
...
-        return onChange('');
-      }
 
-      onChange(id);
-    },
-    [onChange, id, activeId],
+  const setActive = useCallback((): void => {
+    onChange(label);
+  },
+  [onChange, isActive],
```
Comment (MAKARD) [inline src/UI/PickListItem/PickListItem.tsx]: isActive is redundant here

File: src/components/Search/useSearchController.ts [hunk 4]
```diff
@@ -24,7 +24,7 @@ export const useSearchController = (params: UseSearchControllerParams) => {
   }, [params]);
 
   const handleResetSearchInput = useCallback(() => {
-    setSearchTerm('');
+    setSearchTerm({ searchTerm: '' });
```
Comment (MAKARD) [inline src/components/Search/useSearchController.ts]: setSearchOptions

File: src/components/Search/useSearchController.ts [hunk 5]
```diff
@@ -9,13 +9,13 @@ export interface UseSearchControllerParams {
...
   const { searchTerm } = SearchStore.useSearchStore();
-  const setSearchTerm = SearchStore.useSearchTerm();
+  const setSearchTerm = SearchStore.useSetSearch();
 
   const searchInputRef: RefObject<TextInput> = createRef();
 
   const handleChange = useCallback(
     (nextValue: string) => {
-      setSearchTerm(nextValue);
+      setSearchTerm({ searchTerm: nextValue });
```
Comment (MAKARD) [inline src/components/Search/useSearchController.ts]: setSearchOptions

File: src/models.ts [hunk 6]
```diff
@@ -64,3 +64,11 @@ export interface RecipeListModel {
   filters: Array<FilterModel>;
   sortOptions: Array<SortOptionModel>;
 }
+export interface SearchOptions {
```
Comment (MAKARD) [inline src/models.ts]: let's not mix response models and request models.
Keep response models in this file, and move back request model "SearchOption" into recipes.api

File: src/screens/CategoriesList/components/CategoryCards/useCategoryCardsController.ts [hunk 7]
```diff
@@ -7,10 +7,10 @@ export const useCategoryCardsController = () => {
 
   const navigation = useNavigation();
 
-  const setSearchTerm = SearchStore.useSearchTerm();
+  const setSearchTerm = SearchStore.useSetSearch();
 
   const onPress = useCallback((categoryTitle: string) => {
-    setSearchTerm(categoryTitle);
+    setSearchTerm({ searchTerm: categoryTitle });
```
Comment (MAKARD) [inline src/screens/CategoriesList/components/CategoryCards/useCategoryCardsController.ts]: setSearchOptions

File: src/screens/Filter/Filter.tsx [hunk 8]
```diff
@@ -9,23 +9,17 @@ import { useFilterController } from './useFilterController';
...
+        {filters.map(() => {
           return (
-            <FilterItem
-              filter={filter}
-              index={index}
-              key={filter.id}
-              id={filter.id}
-              onChange={onFilterChange}
-            />
+            <FilterItem />
```
Comment (MAKARD) [inline src/screens/Filter/Filter.tsx:22]: key property is missed

File: src/screens/Sort/useSortController.ts [hunk 9]
```diff
@@ -1,33 +1,30 @@
...
+    setSearch({ searchTerm, sort: value });
+    navigation.goBack();
+  },
+  [sortOptions],
   );
 
-  const onSelectPress = () => {
-    navigation.goBack();
-  };
+  useEffect(() => {
```
Comment (MAKARD) [inline src/screens/Sort/useSortController.ts]: redundant code

File: src/stores/search/hooks/useSetSearch.ts [hunk 10]
```diff
@@ -0,0 +1,17 @@
+import * as React from 'react';
```
Comment (MAKARD) [inline src/stores/search/hooks/useSetSearch.ts]: useSetSearch.ts => useSetSearchOptions.ts

File: src/stores/search/searchActions.ts [hunk 11]
```diff
@@ -1,14 +1,16 @@
 import { Dispatch } from 'redux';
 
+import { SearchOptions } from 'src/models';
+
 export enum SearchActions {
   SET = '@search/set',
 }
 
-const actionSetSearchTerm = (searchTerm: string) => ({
+const actionSetSearch = (searchOptions: SearchOptions) => ({
```
Comment (MAKARD) [inline src/stores/search/searchActions.ts]: actionSetSearchOptions

File: src/stores/search/searchActions.ts [hunk 12]
```diff
@@ -1,14 +1,16 @@
 import { Dispatch } from 'redux';
 
+import { SearchOptions } from 'src/models';
+
 export enum SearchActions {
   SET = '@search/set',
```
Comment (MAKARD) [inline src/stores/search/searchActions.ts]: SET_OPTIONS = '@search/set_options

File: src/stores/search/searchActions.ts [hunk 13]
```diff
@@ -1,14 +1,16 @@
...
-const actionSetSearchTerm = (searchTerm: string) => ({
+const actionSetSearch = (searchOptions: SearchOptions) => ({
   type: SearchActions.SET,
-  payload: { searchTerm },
+  payload: searchOptions,
 });
 
-export const setSearchTerm = (searchTerm: string, dispatch: Dispatch) => {
-  dispatch(actionSetSearchTerm(searchTerm));
+export const setSearch = (searchOptions: SearchOptions, dispatch: Dispatch) => {
```
Comment (MAKARD) [inline src/stores/search/searchActions.ts]: setSearchOptions

File: src/stores/search/hooks/index.ts [hunk 14]
```diff
@@ -1,2 +1,2 @@
 export * from './useSearchStore';
-export * from './useSetSearchTerm';
+export * from './useSetSearch';
```
Comment (MAKARD) [inline src/stores/search/hooks/index.ts]: useSetSearchOptions

File: src/screens/Sort/useSortController.ts [hunk 15]
```diff
@@ -1,33 +1,30 @@
...
 import * as RecipesStore from '@stores/recipes';
+import * as SearchStore from '@stores/search';
 
 export const useSortController = () => {
-  const [activeSortId, setActiveSortId] = React.useState('');
+
   const navigation = useNavigation();
 
   const { sortOptions } = RecipesStore.useRecipesStore();
+  const setSearch = SearchStore.useSetSearch();
```
Comment (MAKARD) [inline src/screens/Sort/useSortController.ts]: setSearchOptions

**Review by MAKARD — CHANGES_REQUESTED** [review]

File: src/screens/Filter/Filter.tsx (hunk 8 above)
Comment (valeskus) [reply src/screens/Filter/Filter.tsx:22]: I will realize filter functionality in the future, in another brunch

**Review by MAKARD — APPROVED** [review]

**Pull request merged**